File: src/algorithms/math/binary/Binary.php
<?php
namespace algorithms\math\binary;

/**
 * Converting to binary number.
 * @author DJohnny
 */
class Binary {
	/**
	 * Converts decimal number to binary.
	 * @param int $number Number to convert.
	 * @return string String with binary number.
	 */
	public static function dec2bin($number) {
		return decbin($number);
	}
}
?>

File: src/algorithms/math/binary/binary.dart
import 'dart:html';

void main() {
    querySelector("#num").onInput.listen(binaryEvent);
}

void binaryEvent(Event e) {
    String num = (e.target as InputElement).value;
    try {
        querySelector('#binDart').text=dec2bin(int.parse(num));
    } on FormatException {
        print("Bad format!");
    }
}

/**
 * Converts decimal number to binary using string.
 * @param number Number to convert.
 * @return String with binary number.
 */
String dec2bin(int number) {
    return number.toRadixString(2);
}
